andela/bp-esa-frontend
# #160437221 Add google authentication

#### What does this PR do?
- Login a user with firebase google authentication.
- Logout a user firebase google authentication.
- Protect the routes that are not supposed to be accessed if a user is not logged in.
- Restrict google authentication to `@andela.com` domain.

#### Description of Task to be completed?
- Set up the firebase configuration for google authentication.
- Add signin and signout functions.
- Add regex to restrict login to emails with specific domain.
- Add a custom AuthenticatedRoute for protected routes.

#### How should this be manually tested?
- Clone and set up the project using the instruction in the README.
- Create a `.env` file and add the necessary variables as listed in  `src/firebase/firebase.js`
**NOTE:** Variables are to be provided by Andela's Firebase Admin through Dominic (ESA's devOps engineer).
- Use `npm start` to start the application
- The application should direct you to the login page; when you click login with google, you should be directed to a list of emails.
- Select an andelan email so as to login successfully.

#### What are the relevant pivotal tracker stories?
`#160437221`
